### PR TITLE
Slopen intro

### PR DESCRIPTION
**Pull request by svenjens**
Description 
- what does this feature or fix do?
- please explain (high level) how and why you implemented it this way

Please make sure:
- [ ] to add a label
- [ ] to add one or more reviewers
- [ ] you followed our checklist for [functional testing](https://github.com/Amsterdam/vergunningcheck/blob/master/TESTING.md)
- [ ] you added the necessary documentation (either in the markdown files or inline)
- [ ] you added the necessary automated tests

### CODE REVIEW

File: packages/client/src/intros/shared/Intro.tsx [hunk 1]
```diff
@@ -43,14 +45,18 @@ export default ({
 
   return (
     <>
-      <Paragraph gutterBottom={usableForBullets.length > 0 ? 8 : undefined}>
-        {t("introPage.common.check for permit intro")}{" "}
-        {usableForBullets.length > 0 && t("introPage.common.permit for")}
+      { needReport && <Paragraph>
+        {t("introPage.demolition.need report")}{" "}
+      </Paragraph> }
```
Comment (robinpiets) [inline packages/client/src/intros/shared/Intro.tsx]: My autocorrect is changing these lines. Also, the extra space is not necessary. `{" " }` Do you have "Format document on Save" enabled in your IDE?

File: packages/client/src/intros/shared/Intro.tsx [hunk 2]
```diff
@@ -43,14 +45,18 @@ export default ({
...
+
+      <Paragraph gutterBottom={usableForBullets?.length > 0 ? 8 : undefined}>
+        {!needReport && t("introPage.common.check for permit intro")}{" "}
+        {usableForBullets?.length > 0 && t("introPage.common.permit for")}
```
Comment (robinpiets) [inline packages/client/src/intros/shared/Intro.tsx]: This is wrong, because now this `<Paragraph>` can be rendered with empty content. Can you not do:
`{needReport ? {t("introPage.demolition.need report")} ? t("introPage.common.check for permit intro")}}` ?
If this is not possible, then this paragraph needs conditional rendering as well

File: packages/client/src/intros/shared/Intro.tsx [hunk 3]
```diff
@@ -43,14 +45,18 @@ export default ({
 
   return (
     <>
-      <Paragraph gutterBottom={usableForBullets.length > 0 ? 8 : undefined}>
-        {t("introPage.common.check for permit intro")}{" "}
-        {usableForBullets.length > 0 && t("introPage.common.permit for")}
+      { needReport && <Paragraph>
+        {t("introPage.demolition.need report")}{" "}
```
Comment (robinpiets) [inline packages/client/src/intros/shared/Intro.tsx]: I would change `needReport` to `demolitionChecker` (or something) because that's where the text is from.
`needReport` > `demolition.need report` looks off

**Review by robinpiets — CHANGES_REQUESTED** [review]

File: packages/client/src/i18n/nl.ts [hunk 4]
```diff
@@ -50,12 +50,9 @@ export default {
           "Deze vergunningcheck gaat over kozijnen, deuren, ramen en panelen. U kunt hem gebruiken als u nieuwe plaatst of als u bestaande vernieuwt.",
         "placing bullet": "het plaatsen van de kozijnen zelf.",
       },
-      slopen: {
-        "build without permit exception":
-          "Het gebouw, of het deel van het gebouw wat u gaat slopen, is zonder vergunning gebouwd....?",
-        "intro description":
-          "Deze vergunningcheck gaat over slopen van bouwwerken.",
-        "placing bullet": "het slopen van een bouwwerk zelf.",
+      demolition: {
```
Comment (afjlambert) [inline packages/client/src/i18n/nl.ts]: All other checkers have a dutch name. I like having an English one here. Maybe add to todo we translate the others too?

File: packages/client/src/intros/shared/Intro.tsx (hunk 3 above)
Comment (afjlambert) [reply packages/client/src/intros/shared/Intro.tsx]: This is a generic intro component. Should be reusable and contain no logic for specific checkers. Translations could come from common if really needed. I think needReport is also not the best name. `outcomesIncludeNeedReport` would be better. But the text cannot contain demolition text. Another solution would be to check if the current checker has a key in the translation file, and if so, render the text here.

File: packages/client/src/intros/shared/Intro.tsx [hunk 5]
```diff
@@ -43,14 +45,18 @@ export default ({
...
+      </Paragraph> }
+
+      <Paragraph gutterBottom={usableForBullets?.length > 0 ? 8 : undefined}>
+        {!needReport && t("introPage.common.check for permit intro")}{" "}
+        {usableForBullets?.length > 0 && t("introPage.common.permit for")}
       </Paragraph>
 
-      {usableForBullets.length > 0 && (
+      {usableForBullets?.length > 0 && (
```
Comment (afjlambert) [inline packages/client/src/intros/shared/Intro.tsx:55]: Why did you add a `?`?

File: packages/client/src/intros/shared/Intro.tsx [hunk 6]
```diff
@@ -43,14 +45,18 @@ export default ({
...
         <List data-testid={INTRO_USABLE_FOR_BULLETS} variant="bullet">
-          {usableForBullets.map((bulletText) => (
+          {usableForBullets?.map((bulletText) => (
```
Comment (afjlambert) [inline packages/client/src/intros/shared/Intro.tsx:57]: Why did you add a `?`?

File: packages/client/src/intros/shared/Intro.tsx [hunk 7]
```diff
@@ -69,7 +75,7 @@ export default ({
         </Paragraph>
       )}
 
-      {exceptions.length > 0 && (
+      {exceptions?.length > 0 && (
```
Comment (afjlambert) [inline packages/client/src/intros/shared/Intro.tsx:76]: Why did you add a `?`?

File: packages/client/src/intros/shared/Intro.tsx [hunk 8]
```diff
@@ -82,7 +88,7 @@ export default ({
             style={{ marginBottom: 12 }}
             variant="bullet"
           >
-            {exceptions.map((exception) => (
+            {exceptions?.map((exception) => (
```
Comment (afjlambert) [inline packages/client/src/intros/shared/Intro.tsx:89]: Why did you add a `?`?

File: packages/client/src/intros/shared/Intro.tsx [hunk 9]
```diff
@@ -94,7 +100,7 @@ export default ({
         </>
       )}
 
-      {exceptions.length === 0 && showContactInformation && (
+      {exceptions?.length === 0 && showContactInformation && (
```
Comment (afjlambert) [inline packages/client/src/intros/shared/Intro.tsx:101]: Why did you add a `?`?

File: packages/client/src/intros/shared/Intro.tsx [hunk 10]
```diff
@@ -43,14 +45,18 @@ export default ({
...
     <>
-      <Paragraph gutterBottom={usableForBullets.length > 0 ? 8 : undefined}>
-        {t("introPage.common.check for permit intro")}{" "}
-        {usableForBullets.length > 0 && t("introPage.common.permit for")}
+      { needReport && <Paragraph>
+        {t("introPage.demolition.need report")}{" "}
+      </Paragraph> }
+
+      <Paragraph gutterBottom={usableForBullets?.length > 0 ? 8 : undefined}>
+        {!needReport && t("introPage.common.check for permit intro")}{" "}
```
Comment (afjlambert) [inline packages/client/src/intros/shared/Intro.tsx]: Why don't we show this text in case the outcomes include needReport?
It can still be a combination of both right?

**Review by afjlambert — CHANGES_REQUESTED** [review]

File: packages/client/src/intros/shared/Intro.tsx (hunk 3 above)
Comment (afjlambert) [reply packages/client/src/intros/shared/Intro.tsx]: Please add new props to this component to `Intro.test.tsx`

**Comment by afjlambert** [comment]
I'm missing some context. Can you update the description (with todo's) for this Pull-request?
Maybe it's better to only open the PR if you have all the tests etc., and keep it in draft status until then?

**Review by robinpiets — APPROVED** [review]

**Comment by robinpiets** [comment]
Merged because of deadline
